discovery_engine_core/web-api2/src/server.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+    path::Path,
+};
+
+use clap::Parser;
+use serde::Serialize;
+use tracing::error;
+
+use actix_web::{
+    web::{self, ServiceConfig},
+    App,
+    HttpServer,
+};
+
+use crate::{
+    config::{load_config, Config},
+    middleware::{json_error::wrap_non_json_errors, tracing::tracing_log_request},
+    tracing::init_tracing,
+};
+
+pub(crate) fn default_bind_address() -> SocketAddr {
+    SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 8080).into()
```

Comment:
until now we were using `0.0.0.0`, and maybe we should have different default ports for each service, wdyt?

Reply:
Yes, but I copied the address choice form web-api... 

Generally for testing,local debugging the best default is 127.0.0.x and 0.0.0.0 should be avoided.

But for production you always want 0.0.0.0.

But serde doesn't take additional options and and making the address cfg-depend based on the kind of builds (instead of the context it runs on) is a bit to error prone IMHO. 

Which is why I choose 127.0.0.1 with the expectancy to always explicitly set the address for production usage.